mino/minows/session.go

```diff
@@ -0,0 +1,250 @@
+package minows
+
+import (
+	"context"
+	"encoding/gob"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/rs/zerolog"
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/serde"
+	"golang.org/x/xerrors"
+	"io"
+	"sync"
+)
+
+var ErrWrongAddressType = xerrors.New("wrong address type")
+var ErrNotPlayer = xerrors.New("not player")
+
+// MaxUnreadAllowed Maximum number of unread messages allowed
+// in orchestrator's incoming message buffer before pausing relaying
+const MaxUnreadAllowed = 1e3
+
+type Forward struct {
+	Packet
+	Destination []byte
+}
+
+type orchestrator struct {
+	logger zerolog.Logger
+
+	myAddr orchestratorAddr
+	rpc    rpc
+	// Connects to the participants
+	outs map[peer.ID]*gob.Encoder
```

Comment:
why would we want an encoder per peer ? 🤔

Reply:
Because the gob encoder has a Stream: it's one Stream per peer, and then one gob encoder per Stream.